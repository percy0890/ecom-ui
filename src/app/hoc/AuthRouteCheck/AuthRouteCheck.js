import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import config from 'app/config/index.config';
import AppConstants from 'app/app.constants.json';

export const withAuthCheck = (isPrivate = false) => WrappedComponent => {
  class HOC extends PureComponent {
    componentDidMount() {
      setTimeout(() => {
        if (
          (!this.props.accessToken ||
            this.props.accessToken === AppConstants.INVALID ||
            this.props.accessToken === AppConstants.E403) &&
          isPrivate
        ) {
          // If Access Token doesn't exist or isn't valid and the route being accessed is a private route, redirect to public one
          this.props.history.push(config.PUBLIC_ROOT);
        } else if (
          this.props.accessToken &&
          this.props.accessToken !== AppConstants.INVALID &&
          this.props.accessToken !== AppConstants.E403 &&
          !isPrivate
        ) {
          // If Access Token exists & is valid and the route being accessed is a public route (auth module), redirect to private one (home)
          this.props.history.push(config.PRIVATE_ROOT);
        }
      });
    }

    render() {
      return <WrappedComponent {...this.props} />;
    }
  }

  HOC.propTypes = {
    accessToken: PropTypes.string,
    history: PropTypes.object,
  };

  const mapStateToProps = state => ({
    accessToken: state.global.userInfo.accessToken,
  });

  const withConnect = connect(
    mapStateToProps,
    null,
  );

  return withConnect(HOC);
};
