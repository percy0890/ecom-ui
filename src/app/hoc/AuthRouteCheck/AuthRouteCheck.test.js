import React from 'react';
import { shallow } from 'enzyme';
import { withAuthCheck } from './AuthRouteCheck';
jest.useFakeTimers();
describe('<AuthRouteCheck />', () => {
  it('renders correctly', () => {
    jest.runOnlyPendingTimers();
    expect(shallow(<withAuthCheck accessToken="123" />)).toMatchSnapshot();
  });
});
