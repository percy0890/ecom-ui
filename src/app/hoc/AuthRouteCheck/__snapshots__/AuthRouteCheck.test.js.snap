// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthRouteCheck /> renders correctly 1`] = `
<withAuthCheck
  accessToken="123"
/>
`;
